### PR TITLE
Try multiple cache locations

### PR DESCRIPTION
**Pull request by pawamoy**
In CI environment, depending on the user and whether it has a home, `flakehell` would try to create the `/.cache` directory, and would fail with a `PermissionError`.

This commit changes that by making `flakehell` try to create the cache at different locations: `$HOME/.cache/flakehell` and `./.flakehell_cache` (the chances we can write into the current directory in CI are high).

A `--disable-cache` option would be much more robust but seems to be way more complicated to implement.

### CODE REVIEW

File: flakehell/_logic/_snapshot.py [hunk 1]
```diff
@@ -10,18 +10,36 @@
...
-        return
-    for fpath in path.iterdir():
-        if time() - fpath.stat().st_atime <= THRESHOLD:
-            continue
-        fpath.unlink()
+CACHE_PATHS = (
+    Path.home() / '.cache' / 'flakehell',
+    Path('.') / '.flakehell_cache',
+)
+
```
Comment (pawamoy) [inline flakehell/_logic/_snapshot.py:19]:
```suggestion


```

File: flakehell/_logic/_snapshot.py [hunk 2]
```diff
@@ -10,18 +10,37 @@
...
+
+def prepare_cache():
+    for path in CACHE_PATHS:
+        try:
+            path.mkdir(parents=True, exist_ok=True)
+        except PermissionError as error:
```
Comment (pawamoy) [inline flakehell/_logic/_snapshot.py:25]:
```suggestion
        except PermissionError:
```

**Comment by pawamoy** [comment]
Oh, another solution would be to read the `CACHE_PATH` from an environment variable. Would you prefer that method?

File: flakehell/_logic/_snapshot.py [hunk 3]
```diff
@@ -10,18 +10,37 @@
...
+            path.mkdir(parents=True, exist_ok=True)
+        except PermissionError:
+            pass
+        else:
+            for fpath in path.iterdir():
+                if time() - fpath.stat().st_atime <= THRESHOLD:
+                    continue
+                fpath.unlink()
+            return path
+
```
Comment (pawamoy) [inline flakehell/_logic/_snapshot.py:33]:
```suggestion

```

**Comment by pawamoy** [comment]
I've opened #102 as a replacement, I think it's a simpler and better approach.

**Comment by orsinium** [comment]
Thank you! Let's proceed with #102 then :)

**Pull request closed without merging**